etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2475 @@
+*
+* @name class-methods-use-this
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [class-methods-use-this]{@link https://eslint.org/docs/rules/class-methods-use-this}
+*
+* @example
+* // Bad...
+* class MyClass {
+*   method() {
+*     console.log('no this');
+*   }
+* }
+*
+* // Good...
+* class MyClass {
+*   method() {
+*     console.log(this);
+*   }
+* }
+*/
+rules[ 'class-methods-use-this' ] = 'error';
+
+/**
+* Disallows trailing commas.
+*
+* @name comma-dangle
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [comma-dangle]{@link https://eslint.org/docs/rules/comma-dangle}
+*
+* @example
+* // Okay...
+* const foo = {
+*   bar: 'baz',
+*   qux: 'quux'
+* };
+*/
+rules[ 'comma-dangle' ] = 'off';
+
+/**
+* Verifies calls of `super()` in constructors.
+*
+* @name constructor-super
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [constructor-super]{@link https://eslint.org/docs/rules/constructor-super}
+*
+* @example
+* // Bad...
+* class Bad extends Base {
+*   constructor() {
+*     console.log('missing super()');
+*   }
+* }
+*
+* // Good...
+* class Good extends Base {
+*   constructor() {
+*     super();
+*     console.log('has super()');
+*   }
+* }
+*/
+rules[ 'constructor-super' ] = 'error';
+
+/**
+* Enforces consistent use of curly brace conventions.
+*
+* @name curly
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [curly]{@link https://eslint.org/docs/rules/curly}
+*
+* @example
+* // Bad...
+* if (foo) bar();
+*
+* // Good...
+* if (foo) {
+*   bar();
+* }
+*/
+rules[ 'curly' ] = 'error';
+
+/**
+* Requires a default case in switch statements.
+*
+* @name default-case
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [default-case]{@link https://eslint.org/docs/rules/default-case}
+*
+* @example
+* // Bad...
+* switch (foo) {
+*   case 1:
+*     doSomething();
+* }
+*
+* // Good...
+* switch (foo) {
+*   case 1:
+*     doSomething();
+*     break;
+*   default:
+*     doNothing();
+* }
+*/
+rules[ 'default-case' ] = 'error';
+
+/**
+* Disables the rule for dot notation.
+*
+* @name dot-notation
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [dot-notation]{@link https://eslint.org/docs/rules/dot-notation}
+*
+* @example
+* // Okay...
+* const x = foo[ 'bar' ];
+*/
+rules[ 'dot-notation' ] = 'off';
+
+/**
+* Enforces a newline at the end of files.
+*
+* @name eol-last
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [eol-last]{@link https://eslint.org/docs/rules/eol-last}
+*
+* @example
+* // Bad...
+* const bad = 'no newline';
+* // EOF
+*
+* // Good...
+* const good = 'newline';
+* // EOF
+*/
+rules[ 'eol-last' ] = 'error';
+
+/**
+* Enforces the use of `===` and `!==` in place of `==` and `!=`.
+*
+* @name eqeqeq
+* @memberof rules
+* @type {Array}
+* @default [ 'error', 'always' ]
+* @see [eqeqeq]{@link https://eslint.org/docs/rules/eqeqeq}
+*
+* @example
+* // Bad...
+* if (x == y) {
```

Comment:
@Planeshifter Can you enforce consistent styling? Namely spacing in brackets and parentheses.

Reply:
This applies here and elsewhere.

Reply:
Addressed.